### PR TITLE
Add OIDC config to API server extraArgs

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/roadmap/issues/586

Based on https://github.com/giantswarm/capo-mc-bootstrap/blob/main/helm/capo/templates/kubeadmcontroleplane.yaml#L21-L24

### CODE REVIEW

**Comment by tfussell** [comment]
Tested on `gamma` (in a WC):
```
kubectl get pods -n kube-system kube-apiserver-thom1-default-n68f4 -o yaml
...
    - --oidc-client-id=dex-k8s-authenticator
    - --oidc-groups-claim=groups
    - --oidc-issuer-url=https://dex.thom1.gamma.<redacted>.gigantic.io
    - --oidc-username-claim=email
...
```

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl [hunk 1]
```diff
@@ -14,6 +14,10 @@ spec:
           apiServer:
             extraArgs:
               cloud-provider: external
+              oidc-issuer-url: https://dex.{{ .Values.baseDomain }}
```
Comment (tfussell) [inline helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:17]: It feels kind of weird requiring this to be passed in, but I don't see an easy alternative. If we could access values from `config` then we could default this based on the installation.

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl [hunk 2]
```diff
@@ -14,6 +14,10 @@ spec:
...
+              oidc-client-id: dex-k8s-authenticator
```
Comment (tfussell) [inline helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:18]: Should this be configurable or is this the only allowed value?

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl (hunk 1 above)
Comment (pipo02mix) [reply helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:17]: In the dex app, when it is configured to be deployed in workload cluster we take [the domain from cluster values](https://github.com/giantswarm/dex-app/blob/master/helm/dex-app/templates/dex/secret.yaml#L33). So cluster operator (I think @rossf7 mentioned to me now cluster apps operator would do it in the new CAPI ecosystem) maintain cluster configmap values with
```
> » k get cm int1d-cluster-values -oyaml                                                                                     (giantswarm-flamingo/int1d)
apiVersion: v1
data:
  values: |
    baseDomain: int1d.k8s.flamingo.eu-west-1.aws.gigantic.io
    cluster:
      calico:
        CIDR: 10.237.0.0/16
      kubernetes:
        API:
          clusterIPRange: 10.238.0.0/16
        DNS:
          IP: 10.238.0.10
    clusterCA: |-
      -----BEGIN CERTIFICATE-----
      MIID...
```
can we rely on this on CAPO?

**Review by pipo02mix — APPROVED** [review]
LGTM

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl (hunk 1 above)
Comment (tfussell) [reply helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:17]: There's a chicken and egg problem where the cluster-values config won't be generated until the helm release is created and the helm release won't be created until the config exists. The current idea is to add functionality to organization-operator to create a ConfigMap with installation-specific and organization-specific values such as `baseDomain`. Then we can merge those values into this app. We'll continue on this next week.

**Review by kopiczko — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]
LGTM

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl [hunk 3]
```diff
@@ -14,6 +14,12 @@ spec:
           apiServer:
             extraArgs:
               cloud-provider: external
+              {{- if .Values.oidc.enabled }}
+              oidc-issuer-url: https://dex.{{ include "resource.default.name" $ }}.{{ .Values.baseDomain }}
```
Comment (tfussell) [inline helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:18]: Getting this from the `giantswarm`/`giantswarm-test` appcatalog defaults. See https://github.com/giantswarm/installations/pull/1952 for an example.

**Pull request merged**